ocaml/xapi/xapi_pool_update.ml

```diff
@@ -201,30 +201,14 @@ let extract_update_info ~__context ~vdi ~verify =
         let key = List.assoc "key" attr in
         let uuid = List.assoc "uuid" attr in
         let name_label = List.assoc "name-label" attr in
+        let guidance = List.assoc "after-apply-guidance" attr in
```

Comment:
* Is it possible that `List.assoc` fails or can you be sure that the key `after-apply-guidance` is always present?
* You can compute `guidance` like below. This avoids re-binding `guidance`.

        let guidance =
           List.assoc "after-apply-guidance" attr
           |> String.split
           |> List.map guidance_from_string